src/error.rs

```diff
@@ -56,6 +56,7 @@ use thiserror::Error;
 /// - `NotNaturalNumber` is thrown if the function expects a natural number,
 /// but a number smaller than `1` is provided
 /// - `NotPrime` is thrown if a provided integer is not prime
+/// - `NotSquare` is thrown if a matrix is not square
```

Comment:
```suggestion
/// - `NoSquareMatrix` is thrown if a matrix is not square
```

would be easier to understand